File: app/src/main/java/com/example/dreamloaf/data/ProductDao.java
package com.example.dreamloaf.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;
import java.util.List;

@Dao
public interface ProductDao {
    @Insert
    void insert(Product product);

    @Query("SELECT * FROM products")
    LiveData<List<Product>> getAllProducts();

    @Query("SELECT * FROM products WHERE id = :productId")
    LiveData<Product> getProductById(int productId);

    @Query("DELETE FROM products WHERE id = :productId")
    void delete(int productId);

    @Update
    void update(Product product);
}

File: app/src/main/java/com/example/dreamloaf/data/ProductDao.kt
package com.example.dreamloaf.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface ProductDao {
    @Insert
    fun insert(product: Product)

    @get:Query("SELECT * FROM products")
    val allProducts: LiveData<MutableList<Product?>?>?

    @Query("SELECT * FROM products WHERE id = :productId")
    fun getProductById(productId: Int): LiveData<Product?>?

    @Query("DELETE FROM products WHERE id = :productId")
    fun delete(productId: Int)

    @Update
    fun update(product: Product)
}